### DISCUSSION TITLE
@nx/cypress/plugins/cypress-preset fails with "Error: Web server failed to start in" in Github Actions

### DISCUSSION THREAD

**Discussion by OurxD** [nrwl/nx#26990] [Show and Tell]
## tl:dr
If you're experiencing this issue, set host to `127.0.0.1` on the `ciWebServerCommand`.
```javascript
export default defineConfig({
  e2e: {
    ...nxE2EPreset(__filename, {
      cypressDir: 'src',
      bundler: 'vite',
      webServerCommands: {
        default: 'nx run project-name:serve',
        production: 'nx run project-name:preview',
      },
      ciWebServerCommand: 'npx nx run project-name:serve-static --host 127.0.0.1',
    }),
    baseUrl: 'http://localhost:4200',
  },
});
```

## Long Version
I've been investigating this for a while now. It works locally, ran from my machine as well as in the custom image we use in the action locally, but it fails when it runs in Github actions. It turns out to be due to the node request made to check if the server is running failing with the following:
```javascript
Error: connect ECONNREFUSED 127.0.0.1:8090
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1605:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 4200
}
```
Which hinted the final fix. Even though we're setting the`baseUrl` to be `http://localhost:4200`, the dns resolver ends up calling `127.0.0.1` as mentioned [here](https://github.com/nodejs/node/issues/40702#issuecomment-1100385972), failing to confirm the web server is running. By setting the host to be `127.0.0.1` we ensure it will work in the runner, as node will resolve to that direction anyway. 🤷 

If you have more info around this please don't hesitate to comment or reach out, as I'm more than happy to learn further.
